node-bootstrapper/main.go

```diff
@@ -0,0 +1,396 @@
+	if err != nil {
+		return "", err
+	}
+	return nodeBootstrapping.CSE, nil
+}
+
+// re-implement CustomData + cloud-init logic from AgentBaker
+// only for files not copied during build process.
+func writeCustomData(config *datamodel.NodeBootstrappingConfiguration) error {
+	files, err := customData(config)
+	if err != nil {
+		return err
+	}
+	for path, file := range files {
+		slog.Info(fmt.Sprintf("Saving file %s ", path))
+
+		dir := filepath.Dir(path)
+		if err := os.MkdirAll(dir, 0755); err != nil {
+			return fmt.Errorf("create directory %s: %w", dir, err)
+		}
+		if err := os.WriteFile(path, []byte(file.Content), file.Mode); err != nil {
+			return fmt.Errorf("write file %s: %w", path, err)
+		}
+	}
+	return nil
+}
+
+type File struct {
+	Content string
+	Mode    os.FileMode
+}
+
+func customData(config *datamodel.NodeBootstrappingConfiguration) (map[string]File, error) {
+	contentDockerDaemon, err := contentDockerDaemonJSON(config)
+	if err != nil {
+		return nil, fmt.Errorf("content docker daemon json: %w", err)
+	}
+
+	files := map[string]File{
+		"/etc/kubernetes/certs/ca.crt": {
+			Content: config.ContainerService.Properties.CertificateProfile.CaCertificate,
+			Mode:    ReadOnlyUser,
+		},
+		"/etc/systemd/system/docker.service.d/exec_start.conf": {
+			Content: contentDockerExecStart(config),
+			Mode:    ReadOnlyWorld,
+		},
+		"/etc/docker/daemon.json": {
+			Content: contentDockerDaemon,
+			Mode:    ReadOnlyWorld,
+		},
+		"/etc/default/kubelet": {
+			Content: contentKubelet(config),
+			Mode:    ReadOnlyWorld,
+		},
+	}
+
+	if config.ContainerService.Properties.SecurityProfile.GetPrivateEgressContainerRegistryServer() != "" {
+		files["/etc/containerd/certs.d/mcr.microsoft.com/hosts.toml"] = File{
+			Content: containerDMCRHosts(config),
+			Mode:    ReadOnlyWorld,
+		}
+	}
+
+	if config.EnableSecureTLSBootstrapping || agent.IsTLSBootstrappingEnabledWithHardCodedToken(config.KubeletClientTLSBootstrapToken) {
+		if err := useBootstrappingKubeConfig(config, files); err != nil {
+			return nil, err
+		}
+	} else {
+		useHardCodedKubeconfig(config, files)
+	}
+
+	for path, file := range files {
+		file.Content = strings.TrimLeft(file.Content, "\n")
+		files[path] = file
+	}
+
+	return files, nil
+}
+
+func useHardCodedKubeconfig(config *datamodel.NodeBootstrappingConfiguration, files map[string]File) {
+	files["/var/lib/kubelet/kubeconfig"] = File{
+		Content: contentKubeconfig(config),
+		Mode:    ReadOnlyWorld,
+	}
+}
+
+func useBootstrappingKubeConfig(config *datamodel.NodeBootstrappingConfiguration, files map[string]File) error {
+	bootstrapKubeconfig, err := contentBootstrapKubeconfig(config)
+	if err != nil {
+		return fmt.Errorf("content bootstrap kubeconfig: %w", err)
+	}
+	files["/var/lib/kubelet/bootstrap-kubeconfig"] = File{
+		Content: bootstrapKubeconfig,
+		Mode:    ReadOnlyWorld,
+	}
+	return nil
+}
+
+func contentKubeconfig(config *datamodel.NodeBootstrappingConfiguration) string {
```

Comment:
could we renamed all function starting with `content<artifact>` with maybe something like `generate<artifact>Content` 

When reading the function name I wasn't sure what was happening.